### PR TITLE
Feature/dm 734 sample page with routing

### PR DESCRIPTION
**Pull request by Steffengreiner**
**What has changed**

This PR provides the SamplePage on which an empty view and a filled sample Grid is shown. 
The Sample Information is provided by the sampleInformationService which mocks the fields in a Sample Record shown in the prototype.

### CODE REVIEW

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+  //ToDo is there a java reflection solution to check all fields or should the record implement a solution
+  private boolean isInSample(Sample sample, String searchTerm) {
+    return matchesTerm(sample.id(), searchTerm) || matchesTerm(sample.label(), searchTerm)
+        || matchesTerm(sample.batch(), searchTerm) || matchesTerm(sample.status(), searchTerm)
+        || matchesTerm(sample.experiment(), searchTerm) || matchesTerm(sample.source(), searchTerm)
+        || matchesTerm(sample.condition1(), searchTerm) || matchesTerm(sample.condition2(),
+        searchTerm) || matchesTerm(sample.specimen(), searchTerm) || matchesTerm(sample.species(),
+        searchTerm);
+  }
```
Comment (wow-such-code) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java:307]:
```suggestion
  private boolean isInSample(Sample sample, String searchTerm) {
    boolean result = false;
    for(PropertyDescriptor descriptor : BeanUtils.getPropertyDescriptors(ExperimentalGroupDTO.class)) {
      if(!descriptor.getName().equals("class") {
        // toString() might need to be implemented for useful results
        String value = d.getReadMethod().invoke(bean);
        result |= matchesTerm(value, searchTerm);
    }
    return result;
  }
```

Comment (wow-such-code) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java:307]: this worked for me. need to add the respective imports, of course

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/ProjectViewPage.java [hunk 2]
```diff
@@ -38,30 +39,35 @@ public class ProjectViewPage extends Div implements BeforeEnterObserver, RouterL
...
     handler = new ProjectViewHandler(projectNavigationBarComponent, projectInformationPage,
-        experimentInformationPage);
+        experimentInformationPage, sampleInformationPage);
     log.debug(String.format(
-        "New instance for project view (#%s) created with a project navigation component (#%s), a project information page (#%s) and experiment information page (#%s)",
+        "New instance for project view (#%s) created with a project navigation component (#%s), a project information page (#%s), an experiment information page (#%s), and a sample information page (#%s)",
         System.identityHashCode(this), System.identityHashCode(projectNavigationBarComponent),
         System.identityHashCode(projectInformationPage),
-        System.identityHashCode(experimentInformationPage)));
+        System.identityHashCode(experimentInformationPage),
```
Comment (wow-such-code) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/ProjectViewPage.java:56]: what does this do?

**Review by wow-such-code — CHANGES_REQUESTED** [review]
Very clear for the amount of code, good job!

The aim of the card (routing) works for me. I do have a suggestion for the filter and sorting for the two special fields did not work.

Also NIT: two questions about two of the used classes, if you want to indulge me. Otherwise I can look it up myself. :)

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    buttonAndFieldBar.add(fieldBar, buttonBar);
+    buttonAndFieldBar.setWidthFull();
+  }
+
+  private void createSampleTabs(Map<Experiment, Collection<Sample>> experimentSamples) {
+    experimentSamples.forEach((experiment, samples) -> {
+      Span experimentName = new Span(experiment.getName());
+      Tab experimentSampleTab = new Tab(experimentName);
+      //Todo How to provide information from different services in the vaadin grid?
+      Grid<Sample> sampleGrid = new Grid<>(Sample.class, false);
```
[the remark this answers is not in the document]
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java:141]: NIce catch apparently vaadin column that are generated via a renderer ned a comparator defined to be able to inmemory sort the column values 👁️

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleInformationPage.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ * This page hosts the components necessary to show and update the actual sample information
+ * associated with a provided {@link life.qbic.projectmanagement.domain.project.ProjectId} in the
+ * URL
+ */
+
+@Route(value = "projects/:projectId?/samples", layout = ProjectViewPage.class)
+@SpringComponent
+@UIScope
+@CssImport("./styles/views/project/project-view.css")
+@PermitAll
```
[the remark this answers is not in the document]
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleInformationPage.java:29]: @SpringComponent allows this Page to be autowired into the projectViewPage
@UIScope defines the scope on when the Page will be rebuilt from scratch. (https://vaadin.com/docs/latest/integrations/spring/scopes#uiscope)
@CssImport loads the css defined in the project-view css for this page. This is necessary to allocate the component within the css grid of the project view page 
@PermitAll means that everyone is able to access the page meaning no privileges are needed to do so

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/ProjectViewPage.java (hunk 2 above)
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/ProjectViewPage.java:56]: This logs purpose is to check if the subPages were created. identityHash code ensures that the default hash code for the given object is returned independent of if the given object's class overrides hashCode()

**Review by wow-such-code — APPROVED** [review]
noice

**Pull request merged**